guava/src/com/google/common/primitives/Longs.java

```diff
@@ -247,13 +247,16 @@ public static long constrainToRange(long value, long min, long max) {
    *
    * @param arrays zero or more {@code long} arrays
    * @return a single array containing all the values from the source arrays, in order
+   * @throws IllegalArgumentException if the total number of elements in {@code arrays} does not
+   *     fit in an {@code int}
    */
   public static long[] concat(long[]... arrays) {
-    int length = 0;
+    long length = 0;
     for (long[] array : arrays) {
       length += array.length;
     }
-    long[] result = new long[length];
+    checkNoOverflow(length);
```

Comment:
`Math.toIntExact`?

Reply:
I preferred writing a custom checker to provide
* a better description ("the total number of elements in the arrays must fit in an int" over "integer overflow")
* a better exception type (`IllegalArgumentException` over `ArithmeticException` - I would be really confused seeing `ArithmeticException` after concatenating `long` arrays)

Reply:
(Of course I can replace to the JDK utility if you think it is not worth adding a custom one.)